File: java/src/peachtree/epi/Case.java
package peachtree.epi;

import java.util.HashMap;
import java.util.Set;

public class Case {
	
	
	String accession;
	HashMap<String, String> annotations;
	
	public Case(String accession, HashMap<String, String> annotations) {
		this.accession = accession;
		this.annotations = annotations;
		
		System.out.print("Accession " + accession);
		for (String var : annotations.keySet()) {
			System.out.print("\t" + var + "=" + annotations.get(var) + ".");
		}
		System.out.println();
		
	}

	
	/**
	 * Get the value of this variable
	 * @param var
	 * @return
	 */
	public String getValue(String var) {
		return this.annotations.get(var);
	}


	/**
	 * Get this accession
	 * @return
	 */
	public String getAccession() {
		return this.accession;
	}


	public Set<String> getVariables() {
		return this.annotations.keySet();
	}
	
	

}

File: cpp/peachtree/src/epi/Case.cpp
/*
 * Case.cpp
 *
 *  Created on: 14/07/2021
 *      Author: jdou557
 */

#include "Case.h"

Case::Case(string accession, map<string, string> annotations) {
	this->accession = accession;
	this->annotations = annotations;
}


/*
 * Get the value of this variable
 */
string Case::getValue(string var){
	if (this->annotations.find(var) == this->annotations.end()) return "";
	return this->annotations[var];
}


/*
 * Get this accession
 */
string Case::getAccession(){
	return this->accession;
}


/*
 * Get a list of variables
 */
vector<string> Case::getVariables(){
	vector<string> keys;
	for (const auto & pair : this->annotations){
		keys.push_back(pair.first);
	}
	return keys;
}




void Case::setActive(bool val){
	this->active = val;
}


bool Case::getActive(){
	return this->active;
}
